Staff-Logs: Added Search for Officer and Grade Check for Leaderboard and Search

LeSiiN opened the pull request:
- done with frontend ( backend needed only)
- added LogPerms for Leaderboard and Officer Search with Notify(QBCore)

LeSiiN committed:
added search for officer

added search for officer

LeSiiN committed:
Merge branch 'Project-Sloth:main' into main

LeSiiN committed:
changes for Staff Logs

- done with frontend ( backend needed only)
- added LogPerms for Leaderboard and Officer Search with Notify(QBCore)

xFutte commented on ui/style.css:
```diff
@@ -1729,6 +1729,20 @@ span.tag-input[contenteditable]:empty::before {
+.leaderboard-loader {
+    margin-top: 5vh;
+    margin-left: auto;
+    margin-right: auto;
```
This can actually be refactored to:
```suggestion
    margin: 5vh auto 0 auto;
```

LeSiiN replied:
Did that thank you <3

xFutte commented on ui/style.css:
```diff
@@ -5095,6 +5109,126 @@ span.tag-reports-input[contenteditable]:empty::before {
+    font-size: 18px;
+    font-weight: bolder;
+    padding-top: 5px;
+    padding-bottom: 5px;
```
Can be refactored to `padding: 5px 0;`

LeSiiN replied:
Did that thank you <3

xFutte commented on ui/style.css:
```diff
@@ -5095,6 +5109,126 @@ span.tag-reports-input[contenteditable]:empty::before {
+    text-align: center;
+    margin: auto;
+    margin-top: 10px;
+    margin-bottom: 0px;
```
Refactor to `margin: 10px auto 0 auto`

LeSiiN replied:
Did that thank you <3

xFutte commented on ui/style.css:
```diff
@@ -5095,6 +5109,126 @@ span.tag-reports-input[contenteditable]:empty::before {
+.leaderboard-search-input {
+    background-color: var(--color-3);
+    color: white;
+    padding-top: 10px;
+    padding-bottom: 10px;
```
Refactor to `padding: 10px auto`

LeSiiN replied:
Did that thank you <3

xFutte commented on ui/style.css:
```diff
@@ -5095,6 +5109,126 @@ span.tag-reports-input[contenteditable]:empty::before {
+    padding-top: 10px;
+    padding-bottom: 10px;
+    text-align: center;
+    margin: auto;
+    margin-top: 10px;
+    margin-bottom: 0px;
```
`margin: 10px auto 0 auto;`

LeSiiN replied:
Did that thank you <3

xFutte commented on ui/style.css:
```diff
@@ -5095,6 +5109,126 @@ span.tag-reports-input[contenteditable]:empty::before {
+    width: 95%;
+    color: white;
+    font-size: 15px;
+    background-color: rgba(0, 0, 0, 0);
```
Just use `background-color: transparent`

LeSiiN replied:
Did that thank you <3

xFutte commented on ui/style.css:
```diff
@@ -5095,6 +5109,126 @@ span.tag-reports-input[contenteditable]:empty::before {
+    flex-direction: column;
+    overflow: auto;
+    margin: auto;
+    margin-top: 15px;
```
Combine the three margin properties to `margin: 15px auto 0 auto;`

LeSiiN replied:
Did that thank you <3

xFutte commented on ui/style.css:
```diff
@@ -5095,6 +5109,126 @@ span.tag-reports-input[contenteditable]:empty::before {
+    width: 10vh;
+    height: 10vh;
+    margin-left: 10px;
+    margin-top: auto;
+    margin-bottom: auto;
```
I'm not sure if this is needed.

LeSiiN replied:
Did that thank you <3

xFutte commented on ui/style.css:
```diff
@@ -5095,6 +5109,126 @@ span.tag-reports-input[contenteditable]:empty::before {
+
+.leaderboard-tag {
+    margin: auto;
+    margin-left: 0px;
```
Is this needed? If so, combine the three margin properties to `margin: auto 6px auto 0px`

LeSiiN replied:
Did that thank you <3

LeSiiN committed:
fixed what xFutte found <3

fixed what xFutte found <3

LeSiiN committed:
redo of the last update ( not everything )

- updated search input because of little issue
- image wasnt in the middle anymore

xFutte commented on ui/app.js:
```diff
@@ -1052,6 +1053,42 @@ $(document).ready(() => {
+        $(".leaderboard-search-input").slideUp(250);
+        setTimeout(() => {
+          $(".leaderboard-search-input").css("display", "none");
+        }, 250);
```
If I recall correct, the `slideDown` actually takes a function as a second parameter which triggers, when the animation is done. So you might be able to do something like:
```suggestion
        $(".leaderboard-search-input").slideUp(250, () => {
            $(".leaderboard-search-input").css("display", "none");
        });
```

Try giving it a go. The code would be way more clean that way.

LeSiiN replied:
done <3

xFutte commented on ui/app.js:
```diff
@@ -5736,3 +5774,75 @@ $(".map-clear").on('click', function() {
+      `
+                      <div class="leaderboard-item" data-id="0">
+
+                          <div style="display: flex; flex-direction: column; margin-top: 2.5px; margin-left: 5px; width: 100%; padding: 5px;">
+                          <div style="display: flex; flex-direction: column;">
```
This should be moved into CSS classes.

xFutte commented on ui/app.js:
```diff
@@ -5736,3 +5774,75 @@ $(".map-clear").on('click', function() {
+    let charinfo = value.charinfo;
+    let metadata = value.metadata;
+
+    if (typeof value.charinfo == "string") {
+      charinfo = JSON.parse(charinfo);
+    }
+  
+    if (typeof value.metadata == "string") {
+      metadata = JSON.parse(metadata);
+    }
```
Can `.metadata` return as an integer? Or what do you want to check here?

LeSiiN replied:
code is from profile search so someone had an idea here

xFutte commented on ui/app.js:
```diff
@@ -5736,3 +5774,75 @@ $(".map-clear").on('click', function() {
+      metadata.licences = {};
+    }
+  
+    let name = charinfo.firstname + " " + charinfo.lastname;
```
A better way of doing this to make it more readable would be:

```suggestion
    let name = `${charinfo.firstname} ${charinfo.lastname}`;
```

LeSiiN replied:
done <3

xFutte commented on ui/app.js:
```diff
@@ -5736,3 +5774,75 @@ $(".map-clear").on('click', function() {
+  
+    let name = charinfo.firstname + " " + charinfo.lastname;
+  
+    if (value.pp == '') {
+      value.pp = 'img/not-found.webp'
+    }
```
"pp" is not really that saying of what the is. If possible, I'd suggest to change it to imageUrl.

LeSiiN replied:
pp is profilepic, so i guess thats okay

xFutte commented on ui/app.js:
```diff
@@ -5736,3 +5774,75 @@ $(".map-clear").on('click', function() {
+    leaderboardOfficerHTML += `
+                  <div class="leaderboard-item" data-id="${value.citizenid}">
+                      <img src="${value.pp}" class="leaderboard-image">
+                        <div style="display: flex; flex-direction: column; margin-top: 2.5px; margin-left: 5px; width: 100%; padding: 5px;">
```
The styling here is also better of going into a CSS class.

xFutte commented on server/main.lua:
```diff
@@ -245,6 +245,37 @@ QBCore.Functions.CreateCallback('mdt:server:SearchProfile', function(source, cb,
 end)
 
+QBCore.Functions.CreateCallback('mdt:server:SearchOfficerProfile', function(source, cb, sentData)
+	if not sentData then  return cb({}) end
```
For debugging purposes, something like:

```suggestion
	if not sentData then  
	    print('No data received for searching after officer')
	    return cb({}) 
	end
```

Up to you if you think it makes sense.

LeSiiN replied:
done <3

LeSiiN committed:
little updates ( xFutte <3 )

little updates ( xFutte <3 )

LeSiiN committed:
small update

small update
